src/dmtcpplugin.cpp

```diff
@@ -519,11 +519,18 @@ dmtcp_no_coordinator(void)
 EXTERNC void
```

Comment:
Can we add a _verbose_ comment in front of dmtcp_global_barrier giving the advice that you included in the PR discussion?  Specifically, I'm thinking of something like:
> we only the socket plugin needs a global barrier. All other plugins are dealing with node-local resources (e.g., pids, fds, etc.), and just need node-local barriers. Socket plugin (and IB/RM plugins too) goes beyond a single node and needs to involve coordinator for a global name-service database. A simple thumb rule is to always insert a global-barrier between registering and querying the coordinator name-service database.

Could you word-smith according to make it sufficiently precise, but also to clearly give the intuition to users of DMTCP who didn't find this information elsewhere?  Thanks.